src/enemy.cairo

```diff
@@ -19,29 +25,169 @@ impl EnemySerdeLen of dojo::SerdeLen<Enemy> {
 
 #[system]
 mod spawn_enemies {
+    use traits::Into;
+    use array::Array;
+    use array::ArrayTrait;
+
+    use dojo::world::Context;
+
+    use super::{Enemy, ENEMIES_NB};
+
+    /// Spawn the enemies provided in the array.
+    /// /!\ Panics if the number of enemies provided in the array don't match the expected number
+    /// of enemies defined in the [`suer::ENEMIES_NB`] constant.
+    /// There's no sanity check on any of the properties of the enemies so it's possible to spawn
+    /// enemies with really high speed, on top of each other or too big for the grid.
+    ///
+    /// # Arguments
+    ///
+    /// * `ctx` - Context of the game.
+    /// * `enemies` - Array of enemies. It is assumed that all the enemies in this array are valid.
+    fn execute(ctx: Context, enemies: Array<Enemy>) {
+        let enemies_len = enemies.len();
+        assert(enemies_len == ENEMIES_NB.into(), 'Wrong enemies len provided');
+        let mut i: usize = 0;
+        loop {
+            if i == ENEMIES_NB.into() {
+                break ();
+            }
+            set !(ctx.world, i.into(), (enemies[i]));
```

Comment:
Currently there is a limitation on `set!` where we have to pass in the literal to set
```suggestion
            set !(ctx.world, i.into(), (Enemy {
                ...
            }));
```